aptos-move/block-executor/src/view.rs

```diff
@@ -1516,33 +1617,53 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>> TResourceGroupView for Lat
         resource_tag: &Self::ResourceTag,
         maybe_layout: Option<&Self::Layout>,
     ) -> PartialVMResult<Option<Bytes>> {
-        let mut group_read = self
-            .latest_view
-            .get_resource_group_state()
-            .read_cached_group_tagged_data(
-                self.txn_idx,
-                group_key,
-                resource_tag,
-                maybe_layout,
-                &|value, layout| self.patch_base_value(value, layout),
-            )?;
-
-        if matches!(group_read, GroupReadResult::Uninitialized) {
-            self.initialize_mvhashmap_base_group_contents(group_key)?;
+        self.get_resource_from_group_impl(
+            group_key,
+            resource_tag,
+            UnknownOrLayout::Known(maybe_layout),
+            ReadKind::Value,
+        )
+        .map(|group_read| group_read.expect_value())
+    }
 
-            group_read = self
-                .latest_view
-                .get_resource_group_state()
-                .read_cached_group_tagged_data(
-                    self.txn_idx,
-                    group_key,
-                    resource_tag,
-                    maybe_layout,
-                    &|value, layout| self.patch_base_value(value, layout),
-                )?;
-        };
+    fn resource_size_in_group(
+        &self,
+        group_key: &Self::GroupKey,
+        resource_tag: &Self::ResourceTag,
+    ) -> PartialVMResult<usize> {
+        self.get_resource_from_group_impl(
+            group_key,
+            resource_tag,
+            UnknownOrLayout::Unknown,
+            ReadKind::ResourceSize,
+        )
+        .map(|group_read| {
+            if let GroupReadResult::ResourceSize(maybe_size) = group_read {
+                maybe_size.unwrap_or(0) as usize
+            } else {
+                unreachable!("Group read result must be ResourceSize kind")
+            }
+        })
+    }
 
-        Ok(group_read.into_value().0)
+    fn resource_exists_in_group(
+        &self,
+        group_key: &Self::GroupKey,
+        resource_tag: &Self::ResourceTag,
+    ) -> PartialVMResult<bool> {
+        self.get_resource_from_group_impl(
+            group_key,
+            resource_tag,
+            UnknownOrLayout::Unknown,
+            ReadKind::Exists,
```

Comment:
hmm, is exists in group the same as resource exists? this feels a bit awkward to me since this information doesn't seem derivable from metadata or size. does a resource inside a group even have metadata or size?

Reply:
okay I found how this is implemented, Metadata is unreachable for resource in group

Reply:
Yes! metadata for resource group is the same as if it was a resource, and individual resources in group don't have metadata (they are all modified together as a group).